### PR TITLE
Configuration documentation and client restructuring

### LINKED ISSUE #94: Add documentation for configuration
https://github.com/feathersjs/feathers-configuration is used in the generated application but not documented yet.

### PR DESCRIPTION
**Pull request by daffl**
This pull request adds a Configuration chapter and also reorganizes the client usage a little.

Closes #94

### CODE REVIEW

File: clients/feathers.md [hunk 1]
```diff
@@ -79,4 +79,4 @@ In the browser a client that connects to the local server via websockets can be
 </script>
 ```
 
-In the following chapters we will discuss the different ways to connect to a Feathers server via [REST](rest.md), [Socket.io](socket-io.md) and [Primus](primus.md).
+In the following chapters we will discuss the different ways to use the Feathers as a client via [REST](rest.md), [Socket.io](socket-io.md) and [Primus](primus.md).
```
Comment (ekryski) [inline clients/feathers.md]: I think this wording makes less sense. It makes sense to us but most people have no idea what constitutes a Feathers client.

File: clients/readme.md [hunk 2]
```diff
@@ -1,10 +1,16 @@
...
+Now that we went over how to set up [REST](../rest/readme.md) and [real-time](../real-time/readme.md) APIs on the server we can look at how you can interact with a Feathers server from various clients. 
 
-Even better, Feathers itself can be used as a universal (isomorphic) client. That means that you can easily connect to remote services, register your own client side services and hooks. The Feathers client works in the browser with any front-end framework or in any non-browser JavaScript environment like React Native and other NodeJS servers.
+## Feathers as the client
 
-In this chapter we will talk about the various ways you can talk to a Feathers server, either with the [universal Feathers client](feathers.md) or without.
+Feathers itself can be used as a universal (isomorphic) client. That means that you can easily connect to remote services and register your own client side services and hooks. The Feathers client works in the browser with any front-end framework or in non-browser JavaScript environments like React Native and other NodeJS servers.
```
Comment (ekryski) [inline clients/readme.md]: This is great! :+1:

File: configuration/readme.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
+# Configuration
```
Comment (ekryski) [inline configuration/readme.md]: :+1: :+1:

File: clients/feathers.md (hunk 1 above)
Comment (daffl) [inline clients/feathers.md]: True. I'll change it back. There is a `the` too many in there anyway.

File: clients/readme.md [hunk 4]
```diff
@@ -1,10 +1,16 @@
...
+
+In the [Feathers client chapter](feathers.md) we will talk about how to use Feathers as a client in different environments and get it to connect to a Feathers server via [REST](rest.md), [Socket.io](socket-io.md) and [Primus](primus.md).
+
+## Direct communication
+
+Feathers as the client is not required to use a Feathers API. A Feathers server works great with any client that can connect through HTTP(S) to a REST API or - to also get real-time events - websockets. In the chapter for [direct communication](vanilla.md) we will look at how to directly communicate with a Feathers API via [REST](vanilla-rest.md), [Socket.io](vanilla-socket-io.md) and [Primus](primus.md).
```
Comment (ekryski) [inline clients/readme.md]: I would rephrase to: "In order to communicate with a Feathers API you don't need to use Feathers as the client. A Feathers server..."

Comment (daffl) [inline clients/readme.md]: :+1: It shall be done
